src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs

```diff
@@ -775,7 +797,7 @@ private async ValueTask SendRequestContentAsync(HttpRequestMessage request, Http
         {
             // Wait until we receive a trigger notification that it's ok to continue sending content.
             // This will come either when the timer fires or when we receive a response status line from the server.
-            bool sendRequestContent = await allowExpect100ToContinueTask.ConfigureAwait(false);
+            bool sendRequestContent = await allowExpect100ToContinueTask.ConfigureAwait(false); // TODO: decide what to do about this
```

Comment:
What is the plan for this TODO? Is there an issue number tracking this TODO?

Reply:
Thanks for the keen eye, I totally overlooked this! :+1: 

The underlying `TaskCompletionSource` is either set when a status comes or the `_expect100ContinueTimeout` expires. According to the [actual call site of this method](https://github.com/dotnet/runtime/blob/ecee08f564f7a4c68ff3d81b5ab88e3afecafdc9/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs#L594-L609), when we finally await this, we've either received final status line or we're waiting for the `_expect100ContinueTimeout` to expire, which we cannot do anything about here. IMHO, this particular line will probably remain the same.

But few things I've noticed:
1. We definitely should pass `async` flag here and use it in [// TODO: is async: true correct here?](https://github.com/dotnet/runtime/blob/ecee08f564f7a4c68ff3d81b5ab88e3afecafdc9/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs#L809)
2. We probably should take into account `cancellationToken` and stop the waiting for the timeout if the token is cancelled.

@stephentoub your thoughts on this?

Reply:
Yeah, I think this line probably stays as is.  There's not much we can do about Expect: 100-continue.  If you opt-in to it, you may be blocking one thread doing receives while waiting for the send to asynchronously complete in order for the server to send more response and wake up the receive.  So, I agree everything here should be the same between sync and async.  As for the cancellation token, we're still within the region where the cancellation token has been registered with in SendAsyncCore to dispose of the stream, so I'd expect a stale wait on _expect100ContinueTimeout to become irrelevant, though I wouldn't be surprised if we were to look at Task-related EventSource events if we saw a "your state machine never completed" event.  If we're actually getting into that situation, regardless of sync or async, if we're still waiting for the timer to fire when cancellation is requested, we could update the logic for _expect100ContinueTimeout to also set the TCS when cancellation is requested, by registering with the token again.

Reply:
TODOs removed, `async` passed within and used for `SendRequestContentAsync`.